File: dotnet/Lexy/Lexy.Tests/Specifications/RunLexySpecifications.cs
using Lexy.Compiler.Specifications;
using Microsoft.Extensions.DependencyInjection;
using NUnit.Framework;

namespace Lexy.Poc.Specifications
{
    public class RunLexySpecifications : ScopedServicesTestFixture
    {
        [Test]
        public void AllSpecifications()
        {
            LoggingConfiguration.LogFileNames();

            var runner = GetService<ISpecificationsRunner>();
            runner.RunAll("../../../../../../laws/Specifications");
        }

        [Test]
        public void SpecificFile() // used for debugging a specific file from IDE
        {
            LoggingConfiguration.LogFileNames();

            var runner = GetService<ISpecificationsRunner>();
            //runner.Run("../../../../../../laws/Specifications/Table/LookupValidation.lexy");
            //runner.Run("../../../../../../laws/Specifications/Isolate.lexy");
            //runner.Run("../../../../../../laws/Specifications/Function/Variables.lexy");
            runner.Run("../../../../../../laws/Specifications/BuiltInFunctions/Extract.lexy");
        }
    }
}

File: typescipt/tests/specifications/runLexySpecifications.ts


export class RunLexySpecifications extends ScopedServicesTestFixture {
  it('XXXX', async () => {
   public allSpecifications(): void {
     LoggingConfiguration.LogFileNames();

     let runner = GetService<ISpecificationsRunner>();
     runner.RunAll(`../../../../../../laws/Specifications`);
   }

  it('XXXX', async () => {
   public specificFile(): void // used for debugging a specific file from IDE {
     LoggingConfiguration.LogFileNames();

     let runner = GetService<ISpecificationsRunner>();
// runner.Run(`../../../../../../laws/Specifications/Isolate.lexy`);

     runner.Run(`../../../../../../laws/Specifications/Function/If.lexy`);
     //runner.Run(`../../../../../../laws/Specifications/Function/Variables.lexy`);
     //runner.Run(`../../../../../../laws/Specifications/BuiltInFunctions/Extract.lexy`);
   }
}
